src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+                forkProof: {
+                    forkMilestoneProofs: []
+                },
+                signedBlocks: []
+            },
+            fraudProofs: [],
+            onChainSlashes: [],
+            onChainLatestJoinChannelBlockHash: "",
+            outputStateSnapshotHash: "",
+            exitChannelBlocks: [],
+            disputeAuditingDataHash: "",
+            disputer: this.signerAddress,
+            disputeIndex: 0,
+            previousRecursiveDisputeIndex: ethers.MaxUint256,
+            timeout,
+            selfRemoval: false
+        };
+        return { dispute, timeout };
     }
 
-    //Creates a dispute based on the generated proofs or optimistically timeouts (folds) the provided participant
-    public async createDispute(
-        forkCnt: BigNumberish,
-        foldedParticipant: AddressLike,
-        foldedTransactionCnt: BigNumberish,
-        proofs: ProofStruct[]
-    ): Promise<void> {
-        if (foldedParticipant != NO_PARTICIPANT_TO_FOLD) {
-            console.log("DisputeHandler - createDispute - Timeout");
-        }
+    private async createDisputeStruct(
+        forkCnt: number,
+        transactionCnt: number,
+        proofs: ProofStruct[],
+        timestamp: BigNumberish,
+        timeout?: TimeoutStruct
+    ): Promise<{
+        dispute: DisputeStruct;
+        disputeAuditingData: DisputeAuditingDataStruct;
+    }> {
+        const disputeIndex =
+            await this.stateChannelManagerContract.getDisputeLength(
+                this.channelId
+            );
+        const genesisStateSnapshotHash =
+            this.agreementManager.getForkGenesisSnapshot(
+                forkCnt
+            )!.stateMachineStateHash;
+        const latestStateSnapshotHash = this.agreementManager.getSnapShot(
+            forkCnt,
+            transactionCnt
+        )!.stateMachineStateHash;
+        const stateProof = this.collectStateProof(forkCnt);
+        const onChainSlashes =
+            await this.stateChannelManagerContract.getOnChainSlashedParticipants(
+                this.channelId
+            );
+        const onChainLatestJoinChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const latestExitChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const exitChannelBlocks = this.agreementManager.getExitChannelChain();
 
-        const forkCntNumber = Number(forkCnt);
-        //TODO! stop signing for the current fork
-        this.setForkDisputed(forkCntNumber);
-        proofs.forEach((proof) => this.addProof(forkCntNumber, proof));
-        const _dispute = this.disputes.get(forkCntNumber);
-        if (!_dispute) {
-            await this.createNewDispute(
+        const {
+            encodedModifiedState,
+            exitBlock,
+            totalDeposits,
+            totalWithdrawals
+        } = (await this.stateChannelManagerContract.generateDisputeOutputState(
```

Comment:
First of all 'generateDisputeOutputState' returns 5 arguments and you deconstruct 4. Second this is entirely incorrect since 'generateDisputeOutputState' is not a view or pure function so it returns 'Promise<ethers.ContractTransactionResponse>' as indicated by typescript so when you deconstruct that object (what you're doing) you don't get what you expect to get so everything will be undefined. The way to 'reuse' this function is to deploy it in your local EVM and use it there either by directly interpreting the low level bytes that are returned or by emitting and collecting the event